### PR TITLE
dev-libs/libcgroup: add systemd unit files

### PR DESCRIPTION
**Pull request by lovesegfault**
Closes: https://bugs.gentoo.org/632204
Package-Manager: Portage-2.3.71, Repoman-2.3.17
Signed-off-by: Bernardo Meurer <bernardo@standard.ai>

### CODE REVIEW

**Comment by lovesegfault** [comment]
Pinging @blueness @floppym for review :)

File: dev-libs/libcgroup/files/cgconfig.initd [hunk 1]
```diff
@@ -1,117 +1,52 @@
...
-	while read device mount_point fs_type options other; do
-		if grep -q ${device} <<< ${CGROUP_FS}; then
-			let MAX_INDEX++
-			MOUNT_POINTS[${MAX_INDEX}]=${mount_point}
-			MOUNT_OPTIONS[${MAX_INDEX}]=${options}
-		fi
-	done < ${MOUNTS_FILE}
+CGCONFIGPARSER="/usr/sbin/cgconfigparser"
+CGCLASSIFY="/usr/sbin/cgclassify"
+:	${CG_CONFIGFILE:=/etc/cgroup/cgconfig.conf}
```
Comment (SoapGentoo) [inline dev-libs/libcgroup/files/cgconfig.initd]: turn the tab into a space

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+BDEPEND="
+	sys-devel/bison
+	sys-devel/flex
+	elibc_musl? ( sys-libs/fts-standalone )
+"
+DEPEND="pam? ( virtual/pam )"
+RDEPEND="${DEPEND}"
+
+DOCS=(README_daemon README README_systemd INSTALL)
```
Comment (SoapGentoo) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: is this definition necessary? try removing it and seeing whether `default`/`einstalldocs` catches them all

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+pkg_setup() {
+	local CONFIG_CHECK="~CGROUPS"
+	if use daemon; then
+		CONFIG_CHECK="${CONFIG_CHECK} ~CONNECTOR ~PROC_EVENTS"
+	fi
+	linux-info_pkg_setup
+}
+
+PATCHES=(
```
Comment (SoapGentoo) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: put `PATCHES` before `pkg_setup`

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		$(use_enable tools) \
+		${my_conf}
+}
+
+src_test() {
+	# Use mount cgroup to build directory
+	# sandbox restricted to trivial build,
+	# possible kill Diego tanderbox ;)
+	true
+}
```
Comment (SoapGentoo) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: consider `RESTRICT="test"` instead (with the comment preceding it)

File: dev-libs/libcgroup/files/cgconfig.initd (hunk 1 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/files/cgconfig.initd]: Done

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 3 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Done

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 4 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Good call

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 2 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Done

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 5]
```diff
@@ -12,19 +12,29 @@ SRC_URI="mirror://sourceforge/project/libcg/${PN}/v${PV}/${P}.tar.bz2"
 LICENSE="LGPL-2.1"
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+
```
Comment (mgorny) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: But we don't do empty lines here!

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 6]
```diff
@@ -12,19 +12,29 @@ SRC_URI="mirror://sourceforge/project/libcg/${PN}/v${PV}/${P}.tar.bz2"
...
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-replace-DECLS.patch
+	"${FILESDIR}"/${P}-replace-INLCUDES.patch
+	"${FILESDIR}"/${P}-reorder-headers.patch
+	"${FILESDIR}"/${P}-remove-umask.patch
+)
 
-DOCS=(README_daemon README README_systemd INSTALL)
```
Comment (mgorny) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: I suppose this stops INSTALL from being installed. Installing it sounds weird but maybe this package had a good reason for it?

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 7]
```diff
@@ -76,29 +82,22 @@ src_configure() {
...
 src_install() {
 	default
-	prune_libtool_files --all
+	find "${D}" -name '*.la' -delete || die
 
 	insinto /etc/cgroup
 	doins samples/*.conf || die
 
 	if use tools; then
-		newconfd "${FILESDIR}"/cgconfig.confd-r1 cgconfig || die
```
Comment (mgorny) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: This doesn't seem to belong in this commit.

File: dev-libs/libcgroup/files/cgconfig.confd [hunk 8]
```diff
@@ -1,4 +1,7 @@
 # /etc/conf.d/cgconfig: config file for /etc/init.d/cgconfig
```
Comment (mgorny) [inline dev-libs/libcgroup/files/cgconfig.confd]: Please don't rename those files, it's going to be utterly confusing and make reading `git log` a PITA. Just remove old files if they're no longer used.

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 5 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Fixed

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 6 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: I read through INSTALL and it felt like a pointless file to add, all it does is concern packaging/building.

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+	default
+	prune_libtool_files --all
+
+	insinto /etc/cgroup
+	doins samples/*.conf || die
+
+	if use tools; then
+		newconfd "${FILESDIR}"/cgconfig.confd-r1 cgconfig || die
+		newinitd "${FILESDIR}"/cgconfig.initd-r1 cgconfig || die
+		systemd_dounit "${FILESDIR}"/cgconfig.service || die
```
Comment (mgorny) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: `||die` is not needed here. I get that the original ebuild made this mistake but I don't think you ought to repeat it in the code you added to it.

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild [hunk 10]
```diff
@@ -75,16 +87,9 @@ src_configure() {
...
-	true
-}
-
 src_install() {
 	default
-	prune_libtool_files --all
+	find "${D}" -name '*.la' -delete || die
 
 	insinto /etc/cgroup
 	doins samples/*.conf || die
```
Comment (mgorny) [inline dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Also you want to fix those superfluous dies.

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 9 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Done :)

File: dev-libs/libcgroup/libcgroup-0.41-r6.ebuild (hunk 10 above)
Comment (lovesegfault) [reply dev-libs/libcgroup/libcgroup-0.41-r6.ebuild]: Fixed :)
